GeometricSeqPhenotype.java

```diff
@@ -0,0 +1,308 @@
+        // get new nucleotide sequence after mutating at the random index
+        char[] copyNucleotideSequence = Arrays.copyOf(this.nucleotideSequence, Parameters.startingSequence.length());
+        copyNucleotideSequence[mutationIndexSiteSequence] = mutatedNucleotide;
+
+        // update parameter 4 and 5
+        // count number of epitope and non-epitope mutations
+        int eMutationNew = this.eMutation;
+        int neMutationNew = this.neMutation;
+        if (Biology.SiteMutationVectors.VECTORS.getEpitopeSites().contains(mutationIndexSite)) {
+            eMutationNew += 1;
+        } else {
+            neMutationNew += 1;
+        }
+
+        // uncomment line below to test transition transversion ratio:
+        // TestGeometricSeqPhenotype.mutations.println("" + originalNucleotide + mutatedNucleotide);
+
+        checkRep();
+        Phenotype mutP = new GeometricSeqPhenotype(mutA, mutB, copyNucleotideSequence, eMutationNew, neMutationNew);
+        return mutP;
+    }
+
+    // Returns a vector from the precomputed matrices given wild type and mutant amino acids
+    // and the amino acid site where the nucleotide mutation occurs.
+    private double[] getVectors(int mutationIndexSite, String wildTypeAminoAcid, String mutantAminoAcid) {
+        double mutA = 0.0;
+        double mutB = 0.0;
+        if (!wildTypeAminoAcid.equals(mutantAminoAcid)) {
+            // get indices for the matrix based on the wild type and mutant amino acids
+            // matrix i,j correspond with the String "ACDEFGHIKLMNPQRSTWYV"
+            int mSiteMutationVectors = Parameters.AlphabetType.AMINO_ACIDS.getValidCharacters().indexOf(wildTypeAminoAcid);
+            int nSiteMutationVectors = Parameters.AlphabetType.AMINO_ACIDS.getValidCharacters().indexOf(mutantAminoAcid);
+
+            double[] mutations = Biology.SiteMutationVectors.VECTORS.getVector(mutationIndexSite, mSiteMutationVectors, nSiteMutationVectors); // use matrix at site # where mutation is occurring
+            mutA = getTraitA() + mutations[0]; // r * cos(theta)
+            mutB = getTraitB() + mutations[1]; // r * sin(theta)
+        }
+
+        return new double[]{mutA, mutB};
+    }
+
+    // Mutates nucleotide sequence at given mutationIndexSiteSequence with given char mutatedNucleotide.
+    // Returns the wild type and mutant amino acid as a String[]
+    //
+    // package private helper method so that the updating of the nucleotide sequence can be tested in TestGeometricSeqPhenotype.java
+    String[] mutateHelper(int mutationIndexSiteSequence, char mutatedNucleotide) {
+        int mutationIndexSiteCodon = mutationIndexSiteSequence % 3; // index of nucleotide being mutated in codon {0, 1, 2}
+        int mutationIndexSite = mutationIndexSiteSequence / 3; // site # where mutation is occurring {0, . . ., total number of sites - 1}
+        int codonStartIndex = 3 * mutationIndexSite; // start index of nucleotide being mutated in codon {0, 3, 6, . . .}
```

Comment:
Isn't `codonStartIndex` the same as `mutationIndexSiteSequence`?

Reply:
`mutationIndexSiteSequence` can be any number between 0 - `this.startingSequence.length()`. `codonStartIndex` can only be a multiple of 3. 

**0**, 1, 2 -> **0**
**3**, 4, 5 -> **3**
**6**, 7, 8 -> **6**